From 95b81d2b5ba8d5ae64aeecaee3b8ffe59737ee1c Mon Sep 17 00:00:00 2001
From: Sammy Pfeiffer <Sammy.Pfeiffer@student.uts.edu.au>
Date: Mon, 7 May 2018 18:14:39 +1000
Subject: [PATCH] Fix error: 'cerr' is not a member of 'std'

---
 libqicore/src/logproviderimpl.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libqicore/src/logproviderimpl.cpp b/libqicore/src/logproviderimpl.cpp
index d9315f1..c5c9808 100644
--- a/libqicore/src/logproviderimpl.cpp
+++ b/libqicore/src/logproviderimpl.cpp
@@ -8,6 +8,7 @@
 
 #include <boost/lexical_cast.hpp>
 #include <boost/lockfree/queue.hpp>
+#include <iostream>
 
 #include <qi/anymodule.hpp>
 #include <qi/type/objecttypebuilder.hpp>
-- 
2.7.4

