From 630d15679fe2ce9cc3db75b001d56d3b77c9eb36 Mon Sep 17 00:00:00 2001
From: Sammy Pfeiffer <Sammy.Pfeiffer@student.uts.edu.au>
Date: Mon, 7 May 2018 17:04:40 +1000
Subject: [PATCH] Fix 'cout' is not a member of 'std'

---
 src/iocolor.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/iocolor.cpp b/src/iocolor.cpp
index 17a294b..c98f0cc 100644
--- a/src/iocolor.cpp
+++ b/src/iocolor.cpp
@@ -8,6 +8,7 @@
 #include <qi/iocolor.hpp>
 #include <sstream>
 #include <qi/os.hpp>
+#include <iostream>
 
 #if defined(__APPLE__) or defined(__linux__)
 static std::string makeCol(char c, char modifier = -1) {
-- 
2.7.4

